packages/backend/src/tokens/request.ts

```diff
@@ -56,85 +45,287 @@ function assertSignInUrlFormatAndOrigin(_signInUrl: string, origin: string) {
+      return signedOut(authenticateContext, AuthErrorReason.SessionTokenMissing, '', headers);
+    }
+
+    let verifyResult: JwtPayload;
+
+    try {
+      verifyResult = await verifyToken(sessionToken, authenticateContext);
+    } catch (err) {
+      if (err instanceof TokenVerificationError) {
+        err.tokenCarrier = 'cookie';
+        if (
+          instanceType === 'development' &&
+          (err.reason === TokenVerificationErrorReason.TokenExpired ||
+            err.reason === TokenVerificationErrorReason.TokenNotActiveYet)
+        ) {
+          // This probably means we're dealing with clock skew
+          console.error(
+            `Clerk: Clock skew detected. This usually means that your system clock is inaccurate. Clerk will attempt to account for the clock skew in development.
+
+To resolve this issue, make sure your system's clock is set to the correct time (e.g. turn off and on automatic time synchronization).
+
+---
+
+${err.getFullMessage()}`,
+          );
+
+          // Retry with a generous clock skew allowance (1 day)
+          verifyResult = await verifyToken(sessionToken, { ...authenticateContext, clockSkewInMs: 86_400_000 });
+        }
+      } else {
+        throw err;
+      }
+    }
+
+    return signedIn(authenticateContext, verifyResult!, headers);
+  }
+
+  const pk = parsePublishableKeyWithOptions({
+    publishableKey: options.publishableKey,
+    proxyUrl: options.proxyUrl,
+    domain: options.domain,
+  });
+
+  const instanceType = pk.instanceType;
+
   async function authenticateRequestWithTokenInHeader() {
     try {
-      const state = await runInterstitialRules(ruleOptions, [hasValidHeaderToken]);
-      return state;
+      const verifyResult = await verifyToken(authenticateContext.sessionTokenInHeader!, authenticateContext);
+      return await signedIn(options, verifyResult);
     } catch (err) {
       return handleError(err, 'header');
     }
   }
 
   async function authenticateRequestWithTokenInCookie() {
+    const clientUat = parseInt(authenticateContext.clientUat || '', 10) || 0;
+    const hasActiveClient = clientUat > 0;
+    // TODO rename this to sessionToken
+    const sessionToken = authenticateContext.sessionTokenInCookie;
+    const hasSessionToken = !!sessionToken;
+
+    /**
+     * If we have a handshakeToken, resolve the handshake and attempt to return a definitive signed in or signed out state.
+     */
+    if (handshakeToken) {
+      return resolveHandshake();
+    }
+
+    /**
+     * Otherwise, check for "known unknown" auth states that we can resolve with a handshake.
+     */
+    if (instanceType === 'development' && authenticateContext.derivedRequestUrl.searchParams.has('__clerk_db_jwt')) {
+      const headers = buildRedirectToHandshake();
+      return handshake(authenticateContext, AuthErrorReason.DevBrowserSync, '', headers);
+    }
+
+    if (
+      instanceType === 'production' &&
+      authenticateContext.isSatellite &&
```

Comment:
For increased reliability, consider destructuring `authenticateContext` in line 202.

Reply:
☝️